assets/javascript/apps/pipeline/nodes/widgets.tsx

```diff
@@ -1101,3 +1104,219 @@ function BuiltInToolsWidget(props: WidgetParams) {
     </InputField>
   );
 }
+
+export function TextEditorWidget(props: WidgetParams) {
+  const { parameterValues } = getCachedData();
+  const autocomplete_vars_list: string[] = Array.isArray(parameterValues.text_editor_autocomplete_vars)
+  ? parameterValues.text_editor_autocomplete_vars.map((v: Option) => v.value) : [];
+
+  const modalId = useId();
+  const [isDarkMode, setIsDarkMode] = useState(false);
+  const setNode = usePipelineStore((state) => state.setNode);
+
+  const onChangeCallback = (value: string) => {
+  setNode(
+    props.nodeId,
+    produce((draft) => {
+      draft.data.params[props.name] = value;
+    })
+  );
+};
+
+  const openModal = () =>
+    (document.getElementById(modalId) as HTMLDialogElement)?.showModal();
+
+  useEffect(() => {
+    const updateTheme = () =>
+      setIsDarkMode(document.body.getAttribute("data-theme") === "dark");
+    updateTheme();
+    const observer = new MutationObserver(updateTheme);
+    observer.observe(document.body, { attributes: true });
```

Comment:
The theme attribute is set on `documentElement`
```suggestion
      setIsDarkMode(document.documentElement.getAttribute("data-theme") === "dark");
    updateTheme();
    const observer = new MutationObserver(updateTheme);
    observer.observe(document.documentElement, { attributes: true });
```